docs/dev-guide/plugin-dev.md

```diff
@@ -4,198 +4,91 @@ sidebarDepth: 3
 
-### Generator
+For a 3rd party plugin, the options will be resolved from the prompts or command line arguments when the user executes `vue invoke` (see [Prompts for 3rd party plugins](#prompts-for-3rd-party-plugins)).
 
-A CLI plugin published as a package can contain a `generator.js` or `generator/index.js` file. The generator inside a plugin will be invoked in two possible scenarios:
-
-- During a project's initial creation, if the CLI plugin is installed as part of the project creation preset.
-
-- When the plugin is installed after project's creation and invoked individually via `vue invoke`.
-
-The [GeneratorAPI][generator-api] allows a generator to inject additional dependencies or fields into `package.json` and add files to the project.
-
-A generator should export a function which receives three arguments:
-
-1. A `GeneratorAPI` instance;
+3. The entire preset (`presets.foo`) will be passed as the third argument.
 
-2. The generator options for this plugin. These options are resolved during the prompt phase of project creation, or loaded from a saved preset in `~/.vuerc`. For example, if the saved `~/.vuerc` looks like this:
+### Creating new templates
 
-    ``` json
-    {
-      "presets" : {
-        "foo": {
-          "plugins": {
-            "@vue/cli-plugin-foo": { "option": "bar" }
-          }
-        }
-      }
-    }
-    ```
-
-    And if the user creates a project using the `foo` preset, then the generator of `@vue/cli-plugin-foo` will receive `{ option: 'bar' }` as its second argument.
+When you call `api.render('./template')`, the generator will render files in `./template` (resolved relative to the generator file) with [EJS](https://github.com/mde/ejs).
 
-    For a 3rd party plugin, the options will be resolved from the prompts or command line arguments when the user executes `vue invoke` (see [Prompts for 3rd Party Plugins](#prompts-for-3rd-party-plugins)).
+Let's imagine we're creating [vue-cli-auto-routing](https://github.com/ktsn/vue-cli-plugin-auto-routing) plugin and we want to make the following changes to the project on plugin invoke:
 
-3. The entire preset (`presets.foo`) will be passed as the third argument.
+- create a `layouts` folder with a default layout file;
+- create a `pages` folder with `about` and `home` pages;
+- add a `router.js` to the `src` folder root
 
-**Example:**
+To render this structure, you need to create it first inside the `generator/template` folder:
 
-``` js
-module.exports = (api, options, rootOptions) => {
-  // modify package.json fields
-  api.extendPackage({
-    scripts: {
-      test: 'vue-cli-service test'
-    }
-  })
+![Generator structure](/generator-template.png)
 
-  // copy and render all files in ./template with ejs
-  api.render('./template')
+After template is created, you should add `api.render` call to the `generator/index.js` file:
 
-  if (options.foo) {
-    // conditionally generate files
-  }
-}
+```js
+api.render('./template')
```

Comment:
Let's wrap this in the `module.exports = (api) => {}` function since there is no example of this before :)